### PR TITLE
Add a `sort_fonts` function that internally calls `FCFontSort`.

### PR DESCRIPTION
**Pull request by alokedesai**
* This PR adds a `sort_fonts` function (which internally calls [`FCFontSort`](https://www.freedesktop.org/software/fontconfig/fontconfig-devel/fcfontsort.html)) and an `add_integer` function  (which internally calls [`FCPatternAddInteger`](https://manpages.ubuntu.com/manpages/focal/man3/FcPatternAdd-Type.3.html)).
* The former is especially useful for font fallback support as the information returned by `list_fonts` does not allow for sorting by closeness or trimming results from the font set if the an earlier font in the list already covers the same unicode range as a given font.

### CODE REVIEW

**Comment by alokedesai** [comment]
Hey @wezm 👋🏽 Thanks for maintaining such a fantastic crate. Would love your thoughts on this and whether this is a change you'd accept. Please let me know if you have any questions on the approach!

File: fontconfig/src/lib.rs [hunk 1]
```diff
@@ -511,6 +529,17 @@ pub fn list_fonts<'fc>(pattern: &Pattern<'fc>, objects: Option<&ObjectSet>) -> F
     }
 }
 
+/// Returns a `FontSet` containing fonts sorted by closeness to the supplied `pattern`. If `trim` is true, elements in
+/// the list which don't include Unicode coverage not provided by earlier elements in the list are elided.
```
Comment (wezm) [inline fontconfig/src/lib.rs:526]: Might be worth dropping a link to <https://www.freedesktop.org/software/fontconfig/fontconfig-devel/fcfontsort.html> here.

File: fontconfig/src/lib.rs [hunk 2]
```diff
@@ -511,6 +529,17 @@ pub fn list_fonts<'fc>(pattern: &Pattern<'fc>, objects: Option<&ObjectSet>) -> F
...
+pub fn sort_fonts<'fc>(pattern: &Pattern<'fc>, trim: bool) -> FontSet<'fc> {
+    unsafe {
+        let mut res = sys::FcResultNoMatch;
+        let raw_set = ffi_dispatch!(LIB, FcFontSort, ptr::null_mut(), pattern.pat, trim as c_int /* trim */, ptr::null_mut(), &mut res);
```
Comment (wezm) [inline fontconfig/src/lib.rs]:
```suggestion
    let mut res = sys::FcResultNoMatch; // FcFontSort always returns a (possibly empty) set so we don't need to check this
    let unicode_coverage = ptr::null_mut();
    let config = ptr::null_mut();
    unsafe {
        let raw_set = ffi_dispatch!(LIB, FcFontSort, config, pattern.pat, trim as FcBool, unicode_coverage, &mut res);```

**Review by wezm — CHANGES_REQUESTED** [review]
Overall the changes look good. Some minor tweaks suggested. Also please make sure to run `cargo fmt`.

**Comment by alokedesai** [comment]
Thanks for the review, PTAL.

File: fontconfig/src/lib.rs (hunk 1 above)
Comment (alokedesai) [reply fontconfig/src/lib.rs:526]: done.

**Review by wezm — APPROVED** [review]

**Pull request merged**